web/static/js/components/simulation/QuestionnaireSimulation.jsx

```diff
@@ -0,0 +1,169 @@
+// @flow
+import React, { Component } from 'react'
+import { withRouter } from 'react-router'
+import * as questionnaireActions from '../../actions/questionnaire'
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
+import { startSimulation, messageSimulation } from '../../api.js'
+import ChatWindow from './ChatWindow'
+import DispositionChart from './DispositionChart'
+import SimulationSteps from './SimulationSteps'
+
+type ChatMessage = {
+  type: string,
+  body: string
+}
+
+type Submission = {
+  stepId: string,
+  response: ?string
+}
+
+type Simulation = {
+  messagesHistory: Array<ChatMessage>,
+  submissions: Array<Submission>,
+  simulationStatus: string,
+  disposition: string,
+  respondentId: string,
+  currentStep: string,
+  questionnaire: Questionnaire
+}
+
+type Props = {
+  projectId: number,
+  questionnaireId: number,
+  mode: string,
+  questionnaireActions: Object
+}
+
+type State = {
+  simulation: ?Simulation
+}
+
+class QuestionnaireSimulation extends Component<Props, State> {
+  state = {
+    simulation: null
+  }
+
+  fetchQuestionnaireForTitle = () => {
+    const { projectId, questionnaireId } = this.props
+    // TODO: this component doesn't do this fetch for itself
+    // It does it just because the title component above needs the questionnaire being in the redux store to render properly
+    // But for some reason that component isn't able to get the projectId and questionnaireId params properly from the url
+    this.props.questionnaireActions.fetchQuestionnaireIfNeeded(projectId, questionnaireId)
```

Comment:
Aren't you missing a `withRouter` in the title component?

Reply:
I'm almost sure it wasn't because of that, but I'll try it again and let you know! Thanks!

Reply:
OK, yes, adding `withRouter` I was able to access the params.

Now, it's solved [here](https://github.com/instedd/surveda/pull/1698/commits/40b62e16f91fc789f5458a87bcfc0bb2b2559922). Thanks!!!